apps/admin-ui/src/component/BirthDate.tsx

```diff
@@ -1,41 +1,79 @@
 import React, { useState } from "react";
-import { formatDate } from "../common/util";
-import { HorisontalFlex } from "../styles/layout";
+import { gql, useQuery } from "@apollo/client";
+import styled from "styled-components";
+import { useTranslation } from "next-i18next";
+import type { Query, QueryUserArgs } from "common/types/gql-types";
+import { formatDate } from "@/common/util";
+import { HorisontalFlex } from "@/styles/layout";
 
 type Props = {
-  dateOfBirth: string | undefined;
-  showLabel: string;
-  hideLabel: string;
+  userPk: number;
 };
 
-export const BirthDate = ({
-  dateOfBirth,
-  hideLabel,
-  showLabel,
-}: Props): JSX.Element => {
+// NOTE separate query because all requests for dateOfBirth are logged
+// so don't make them automatically or inside other queries
+const DATE_OF_BIRTH_QUERY = gql`
+  query getDateOfBirth($pk: Int!) {
+    user(pk: $pk) {
+      pk
+      dateOfBirth
+    }
+  }
+`;
+
+const Button = styled.button`
+  margin: 0;
+  padding: 0;
+  border: none;
+  background: none;
+  color: var(--color-bus);
+  text-decoration: underline;
+`;
+
+/// Component for toggling the visibility of the user's birth date
+/// @param userPk - pk of the user
+/// Only makes the query if the user clicks the show button to minimise logging
+export function BirthDate({ userPk }: Props): JSX.Element {
   const [visible, setVisible] = useState(false);
 
+  const {
+    data,
+    loading: isLoading,
+    error,
+  } = useQuery<Query, QueryUserArgs>(DATE_OF_BIRTH_QUERY, {
+    variables: {
+      pk: userPk,
+    },
+    fetchPolicy: "network-only",
+    skip: !userPk || !visible,
+  });
+
+  const { t } = useTranslation();
+
+  const user = data?.user ?? undefined;
```

Comment:
`?? undefined` seems redundant

Reply:
It's because GQL types from codegen are Maybe<T> and Maybe !== undefined. So TS is going to cry about it, while in JS they work the same.

Reply:
But in this case it's not needed. Habit of putting it there because it's needed when passing to other components with type signature `var?: T`.